elastic/kibana
# [unified search] fix Changing label of a geospatial filter on causes filter disappear from map

Fixes https://github.com/elastic/kibana/issues/107044, https://github.com/elastic/kibana/issues/109340, and https://github.com/elastic/kibana/issues/136231

### Background
In Maps, users can [create spatial filters](https://www.elastic.co/guide/en/kibana/current/maps-create-filter-from-map.html). The filter is added to the Filter bar.
<img width="300" alt="Screen Shot 2021-09-03 at 4 07 33 PM" src="https://user-images.githubusercontent.com/373691/133142628-9b03dec2-69ee-4c7c-ac8d-841168c88a94.png">
There are 2 important properties about spatial filters created from a Map:
* `filter.meta.type` is set to `spatial_filter`. Maps uses `filter.meta.type` to identify spatial filters. Spatial filters are displayed on the map to provide a visual reference of all applied spatial filters.
* `filter.meta.isMultiIndex` is set to true since Elasticsearch Query DSL for the filter is not tied to a single index pattern or field. The Elasticsearch Query DSL is generated in such a way that the filter works for all map layers, regardless of index pattern or spatial field. In the screenshot above a single filter is filtering both layers, even though they are from different index patterns and different spatial fields.
  * Each layer identifies one or more geo_point or geo_shape fields that is used to display features for that layer
  * The Elasticsearch Query DSL combines these into a `bool.should` clause where each field provides a `bool.must` clause ensuring the spatial field exists and that the document satisfies the spatial filter. Any document matching any of the `bool.must` clauses allows the bool.should clause to produce a match.

### Solution

[Earlier solution](https://github.com/elastic/kibana/pull/111897) proposed Sept 2021. This original effort was abandoned as priorities shifted. I investigated re-opening original solution, which included a filter operator registry. This solution is no longer appropriate with `combined` filters.

Instead, this PR takes a less complex approach and when `filter.meta.isMultiIndex`, the filter editor displays the following:
* Only displays "custom" editor. "Values" editor is not available
* "Edit as filter values" toggle button is not visible
* Data view select is not visible
* onSubmit does not change filter to "custom" unless DSL is edited

<img width="500" alt="Screen Shot 2023-03-30 at 10 33 58 AM" src="https://user-images.githubusercontent.com/373691/228918145-61ddda8b-43a3-44e8-8316-a803473f5a36.png">

### Test instructions
* Install web logs sample data set and open "[Logs] Total Requests and Bytes" map
* Click wrench and select "Draw bounds to filter data". Draw bounds and create a filter.
* In the Filter bar, Click "Edit filter". 
* Change the label and click save. The filter should still be displayed on the map and the filter pill should have a new label.


<strike>Note for testing, until https://github.com/elastic/kibana/pull/153816 is merged, spatial filters with a single geofield will not set `isMultiIndex` to true and will still have the errors referenced in the issues. Once https://github.com/elastic/kibana/pull/153816 is merged all spatial filters will set isMultiIndex.</strike>
